src/basic/copy.c

```diff
@@ -132,6 +130,11 @@ int copy_bytes(int fdf, int fdt, uint64_t max_bytes, bool try_reflink) {
                         assert(max_bytes >= (uint64_t) n);
                         max_bytes -= n;
                 }
+                /* sendfile accepts at most SSIZE_MAX-offset bytes to copy,
+                 * so reduce our maximum by the amount we already copied,
+                 * but don't go below our copy buffer size, unless we are
+                 * close the the limit of bytes we are allowed to copy. */
+                m = MAX(MIN(COPY_BUFFER_SIZE, max_bytes), m - n);
```

Comment:
This would be replaced by `CLAMP(max_bytes, m - n, COPY_BUFFER_SIZE)`. But `m - n > COPY_BUFFER_SIZE` usually, and it's not obvious what `CLAMP` does in that case. I'd rather not depend on the specific order of evaluation in `CLAMP`.